--- pcl-pcl-1.7.1/CMakeLists.txt~	2014-06-03 17:17:51.590108959 +0800
+++ pcl-pcl-1.7.1/CMakeLists.txt	2014-06-03 17:17:57.920108981 +0800
@@ -247,7 +247,9 @@
 endif(LIBUSB_1_FOUND)
 
 # OpenNI
-find_package(OpenNI)
+if(WITH_OPENNI)
+  find_package(OpenNI)
+endif()
 if (OPENNI_FOUND)
   set(HAVE_OPENNI ON)
   include_directories(SYSTEM ${OPENNI_INCLUDE_DIRS})
@@ -292,7 +294,9 @@
 endif()
 
 # Find VTK
-find_package(VTK)
+if(WITH_VTK)
+  find_package(VTK)
+endif()
 if(VTK_FOUND)
   if (PCL_SHARED_LIBS OR 
       (NOT (PCL_SHARED_LIBS) AND NOT (VTK_BUILD_SHARED_LIBS)))
