diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0775119..c7479f1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -208,6 +208,7 @@ if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
     endif ()
 endif()
 
+if (NOT LIB_INSTALL_DIR OR NOT LIB_ARCH_INSTALL_DIR OR NOT DOC_INSTALL_DIR)
 if (WIN32 OR APPLE)
     # On Windows/MacOSX, applications are usually installed on a directory of
     # their own
@@ -223,6 +224,7 @@ else ()
         set (LIB_ARCH_INSTALL_DIR lib/${CMAKE_PROJECT_NAME})
     endif ()
 endif ()
+endif ()
 
 set (SCRIPTS_INSTALL_DIR ${LIB_INSTALL_DIR}/scripts)
 set (WRAPPER_INSTALL_DIR ${LIB_ARCH_INSTALL_DIR}/wrappers)
@@ -353,7 +355,9 @@ add_subdirectory (dispatch)
 add_subdirectory (helpers)
 add_subdirectory (wrappers)
 add_subdirectory (image)
-add_subdirectory (retrace)
+if (ENABLE_CLI OR ENABLE_GUI)
+	add_subdirectory (retrace)
+endif ()
 
 
 ##############################################################################
