--- /tmp/qttools-opensource-src-5.0.1/src/qtconfig/mainwindow.cpp	2013-01-30 03:03:28.000000000 +0800
+++ src/qtconfig/mainwindow.cpp	2013-02-13 22:16:38.229898632 +0800
@@ -61,7 +61,6 @@
 #include <QMessageBox>
 #include <QStyle>
 #include <QtEvents>
-#include <QInputContext>
 #include <QtDebug>
 #include <QPixmap>
 
@@ -752,7 +751,7 @@
     if (row > ui->substitutionsListBox->count())
         row = ui->substitutionsListBox->count() - 1;
     ui->substitutionsListBox->setCurrentRow(row);
-    QFont::removeSubstitution(ui->familySubstitutionCombo->currentText());
+    QFont::removeSubstitutions(ui->familySubstitutionCombo->currentText());
     QFont::insertSubstitutions(ui->familySubstitutionCombo->currentText(), subs);
     setModified(true);
 }
@@ -791,7 +790,7 @@
     ui->substitutionsListBox->clear();
     ui->substitutionsListBox->insertItems(0, subs);
     ui->substitutionsListBox->setCurrentRow(row + 1);
-    QFont::removeSubstitution(ui->familySubstitutionCombo->currentText());
+    QFont::removeSubstitutions(ui->familySubstitutionCombo->currentText());
     QFont::insertSubstitutions(ui->familySubstitutionCombo->currentText(), subs);
     setModified(true);
 }
@@ -809,7 +808,7 @@
     ui->substitutionsListBox->clear();
     ui->substitutionsListBox->insertItems(0, subs);
     ui->substitutionsListBox->setCurrentRow(row - 1);
-    QFont::removeSubstitution(ui->familySubstitutionCombo->currentText());
+    QFont::removeSubstitutions(ui->familySubstitutionCombo->currentText());
     QFont::insertSubstitutions(ui->familySubstitutionCombo->currentText(), subs);
     setModified(true);
 }
