Description: Put frameworks below /usr/share
Author: Deepak Tripathi <apenguinlinux@gmail.com>
Last-Update: 2011-07-03

--- a/lib/compass/frameworks.rb
+++ b/lib/compass/frameworks.rb
@@ -3,7 +3,7 @@
     extend self
 
     ALL = []
-    DEFAULT_FRAMEWORKS_PATH = File.join(Compass.base_directory, 'frameworks')
+    DEFAULT_FRAMEWORKS_PATH = File.join('/usr/share/compass', 'frameworks')
 
     class Framework
       attr_accessor :name
